src/core/Ros.js

```diff
@@ -56,7 +57,17 @@ Ros.prototype.__proto__ = EventEmitter2.prototype;
  * @param url - WebSocket URL for Rosbridge
  */
 Ros.prototype.connect = function(url) {
-  this.socket = assign(new WebSocket(url), socketAdapter(this));
+  if( typeof(io) !== 'undefined' && this.socketio){
+    this.socket = assign(io(url, {'force new connection': true}), socketAdapter(this));
+
+    this.socket.on('connect', this.socket.onopen);
+    this.socket.on('data', this.socket.onmessage);
+    this.socket.on('close', this.socket.onclose);
+    this.socket.on('error', this.socket.onerror);
+  }else{
```

Comment:
Style note, looks like we use a whitespace in here: `} else {` and in `if () {` as in `if (options.url) {`